src/ledger/LedgerManagerImpl.h

```diff
@@ -84,6 +84,7 @@ class LedgerManagerImpl : public LedgerManager
         std::vector<TransactionFrameBasePtr> const& txs,
         std::vector<MutableTxResultPtr> const& mutableTxResults,
         AbstractLedgerTxn& ltx, TransactionResultSet& txResultSet,
+        std::optional<TransactionResultSet> const& expectedResults,
```

Comment:
This function now takes three different results structures, making it more difficult to reason about. If i understand correctly:
* mutableTxResults is the initial result returned by `processFeesSeqNums`, this functions will copy these and maybe modify them
* Final results are appended to txResultSet (these are hashed into the ledger, so have to be in valid state)
* expectedResults are final results form archives that will override txResultSet

Should `expectedResults` live inside MutableTxResult? This way you can (a) avoid mutable state inside tx frame classes, (b) skip application if mutable tx result already contains "final" results, (c) remove this extra parameter from the function signature.